src/main/java/ui/UI.java

```diff
@@ -239,6 +242,30 @@ private void initUI(Stage stage) {
         loadFonts();
         String css = initCSS();
         applyCSS(css, scene);
+
+        setApplicationIcon(stage);
+    }
+
+    public void setApplicationIcon(Stage stage) {
+        stage.getIcons().add(new Image(UI.class.getResourceAsStream("logo.png")));
```

Comment:
Aside from setting dock icon for Windows, this bit will set up the icon on the title bar of the application as shown in the image below, for both Windows and Mac. So yes, this bit is needed for OSX.
![screen shot 2015-12-26 at 11 06 19 am](https://cloud.githubusercontent.com/assets/3381536/12004989/f035d402-abc0-11e5-8e93-693cd737c12a.png)

Alright, I will pull out the image file name into a constant.